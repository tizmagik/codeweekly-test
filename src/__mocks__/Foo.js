import React from 'react';

export default () => {
  console.log('Mocked foo!');
  return <div>Mocked Foo!</div>;
}
