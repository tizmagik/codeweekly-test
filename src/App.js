import React from 'react';
import Foo from './Foo';

const App = () => (
  <main>
    <Foo />
  </main>
);

export default App;
