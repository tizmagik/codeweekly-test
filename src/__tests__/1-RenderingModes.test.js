import React from 'react';
import {mount, render, shallow} from 'enzyme';
import App from '../App';
import Foo from '../Foo';

describe('App', () => {
  it('as mounted', () => {
    expect(mount(<App />)).toMatchSnapshot();
  });

  it('as rendered', () => {
    expect(render(<App />)).toMatchSnapshot();
  });

  it('as shallow', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});

describe('Enzyme', () => {
  // We'll use mount() to give us the fullest representation
  const wrapper = mount(<App />);

  it('find', () => {
    // Find 'Foo' component using Component Name
    expect(wrapper.find('Foo'));

    // Or by using the component constructor
    expect(wrapper.find(Foo));

    // Could also find by .classNames, but that's brittle and not recommended
  });
})
