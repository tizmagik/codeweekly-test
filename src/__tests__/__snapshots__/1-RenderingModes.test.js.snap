exports[`App as mounted 1`] = `
<App>
  <main>
    <Foo>
      <div>
        Foo
      </div>
    </Foo>
  </main>
</App>
`;

exports[`App as rendered 1`] = `
<main>
  <div>
    Foo
  </div>
</main>
`;

exports[`App as shallow 1`] = `
<main>
  <Foo />
</main>
`;
